python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -105,43 +108,136 @@ class DgPartialConfig(TypedDict, total=False):
-        raise DgValidationError("`verbose` must be a boolean.")
-    if "builtin_component_lib" in config and not isinstance(config["builtin_component_lib"], str):
-        raise DgValidationError("`builtin_component_lib` must be a string.")
-    if "use_dg_managed_environment" in config and not isinstance(
-        config["use_dg_managed_environment"], bool
-    ):
-        raise DgValidationError("`use_dg_managed_environment` must be a boolean.")
-    if "require_local_venv" in config and not isinstance(config["require_local_venv"], bool):
-        raise DgValidationError("`require_local_venv` must be a boolean.")
-    if "component_package" in config and not isinstance(config["component_package"], str):
-        raise DgValidationError("`component_package` must be a string.")
-    if "component_lib_package" in config and not isinstance(config["component_lib_package"], str):
-        raise DgValidationError("`component_lib_package` must be a string.")
-    if "is_code_location" in config and not isinstance(config["is_code_location"], bool):
-        raise DgValidationError("`is_code_location` must be a boolean.")
-    if "is_component_lib" in config and not isinstance(config["is_component_lib"], bool):
-        raise DgValidationError("`is_component_lib` must be a boolean.")
-    if "is_deployment" in config and not isinstance(config["is_deployment"], bool):
-        raise DgValidationError("`is_deployment` must be a boolean.")
+
+    _normalize_config_entry(config, "disable_cache", bool)
+    _normalize_config_entry(config, "cache_dir", (str, Path), as_path=True)
+    _normalize_config_entry(config, "verbose", bool)
+    _normalize_config_entry(config, "builtin_component_lib", str)
+    _normalize_config_entry(config, "use_dg_managed_environment", bool)
+    _normalize_config_entry(config, "require_local_venv", bool)
+    _normalize_config_entry(config, "component_package", str)
+    _normalize_config_entry(config, "component_lib_package", str)
+    _normalize_config_entry(config, "is_code_location", bool)
+    _normalize_config_entry(config, "is_component_lib", bool)
+    _normalize_config_entry(config, "is_deployment", bool)
+
+    if "code_locations" in config:
+        if not isinstance(config["code_locations"], list):
+            raise DgValidationError("`code_locations` must be a list.")
+        normalized_code_location_specs = []
+        for spec in config["code_locations"]:
+            if not isinstance(spec, dict):
+                raise DgValidationError("`code_locations` must be a list of dictionaries.")
+            normalized_code_location_specs.append(_normalize_code_location_spec(spec))
+        config["code_locations"] = normalized_code_location_specs
 
     if unrecognized_keys := [k for k in config.keys() if k not in DgPartialConfig.__annotations__]:
         raise DgValidationError(f"Unrecognized fields in configuration: {unrecognized_keys}")
     return cast(DgPartialConfig, config)
 
 
+def _normalize_config_entry(
```

Comment:
Would prefer a function that returns the modifying object and puts the onus on the caller to assign the value

Reply:
you could also make a specific "normalize to path" function that takes the entire dict and returns a new dict since that is the only thing that modifies